support skaffold as a development env on Mac OS

steved opened the pull request:
Due to underlying library limitations, Forge doesn't support local development. Skaffold allows us to do the iteration cycle in a local minikube cluster based on Virtualbox.

Usage:
```
$ e2e/dependencies.sh
$ skaffold dev
[dev - run iteration loop]

$ skaffold debug --port-forward
[connect using local IDE / delve]
```

Better documentation _WIP_ before merging.

sonnysideup commented on controllers/containerimagebuild_controller.go:
```diff
@@ -118,6 +118,7 @@ func (r *ContainerImageBuildReconciler) Reconcile(req ctrl.Request) (ctrl.Result
 
 		build.Status.SetState(forgev1alpha1.Failed)
 		build.Status.ErrorMessage = err.Error()
+		build.Status.BuildCompletedAt = &metav1.Time{Time: time.Now()}
```
I guess it makes sense to set the completed at datetime after a failure. Good catch.

sonnysideup reviewed (approved):
Looks good. I would just add a quick "Development" section to the README.

steved commented:
@sonnysideup might be worth a second pass. Might also have to wait for #42 before I merge it.

steved committed:
support skaffold as a development env on Mac OS

Due to underlying library limitations, Forge doesn't support local development. Skaffold allows us to do the iteration cycle in a local minikube cluster based on Virtualbox.

Usage:
```
$ e2e/dependencies.sh
$ skaffold dev
[dev - run iteration loop]

$ skaffold debug --port-forward
[connect using local IDE / delve]
```

steved committed:
remove fuse3 from Dockerfile.dev

steved committed:
fix e2e/run-test.sh

Commit:
fix endquote in readme

steved committed:
fix build-dev CI job

Commit:
adding kustomize as a prerequisite within the README

Commit:
Changing helm install -> helm upgrade for docker-registry-2

just to match the other ones + ran into some installation issues with how it was
also added a completion message just so folks know
